src/ConnectCore.js

```diff
@@ -137,14 +137,15 @@ class ConnectCore {
   }
 
   // TODO support contract.new (maybe?)
-  contract (abi, uriHandler = this.uriHandler) {
-    const self = this
-    const txObjectHandler = (methodTxObject) => self.txObjectHandler(methodTxObject, uriHandler)
+  contract (abi) {
```

Comment:
yeah you may be, but my original intention as that you can pass a uriHandler for each contract call. Basically you pass a uriHandler for each request. Before this change you could only set the uriHandler once for the contract object, then it was the same for each request with that contract object.